internal/app/service/notification/service/notification.go

```diff
@@ -0,0 +1,58 @@
+package service
+
+import (
+	"proctor/internal/app/service/infra/logger"
+	"proctor/internal/app/service/infra/plugin"
+	"proctor/pkg/notification"
+	"proctor/pkg/notification/event"
+	"sync"
+)
+
+type NotificationService interface {
+	Notify(evt event.Event)
+}
+
+type notificationService struct {
+	observers           []notification.Observer
+	goPlugin            plugin.GoPlugin
+	pluginsBinary       []string
+	pluginsExportedName []string
+	once                sync.Once
+}
+
+func (s *notificationService) Notify(evt event.Event) {
+	s.initializePlugin()
+	for _, observer := range s.observers {
```

Comment:
is it possible if the Event has a type? for now we only implement ExectionEvent, how about if we implement other events Such as ExecutionFinishedEvent? 
And probably user will register some observer for ExecutionEvent, and other observers for ExecutionFinishedEvent?